src/zpivchain.cpp

```diff
@@ -392,3 +394,15 @@ std::list<libzerocoin::CoinDenomination> ZerocoinSpendListFromBlock(const CBlock
     return vSpends;
 }
 
+int64_t GetZerocoinSupply()
+{
+    if (mapZerocoinSupply.empty())
```

Comment:
Something that have in mind, as now `mapZerocoinSupply` works under an specific context (tip), we should add an `AssertLockHeld(cs_main);` here and  double check that every line viewing/modifying `mapZerocoinSupply` is previously locking `cs_main`.

Reply:
sure